app/models/foreman_tasks/task/dynflow_task.rb

```diff
@@ -118,7 +118,8 @@ def main_action
                else
                  execution_plan_action.input
                end
-        @main_action = active_job_action(args['job_class'], args['job_arguments'])
+        args = args['job_data'] || args.slice('job_class')
```

Comment:
For the tasks created before this change, shouldn't this be:

```ruby
args = args['job_data'].try { |data| data['arguments'] } || args['job_arguments']
@main_action = active_job_action(args['job_class'], args['arguments'])
```

Reply:
Yes, you are right :+1: thanks.

Reply:
One small ask: please add a comment that we need the fallback for `job_arguments` for fobs from Dynflow `<=1.2.1`, for the future version of us that would wonder why we're doing this here :)